k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java

```diff
@@ -1255,15 +1269,39 @@ private void processMessageToForward(MessageViewInfo messageViewInfo) throws Mes
         if (!TextUtils.isEmpty(message.getMessageId())) {
             mInReplyTo = message.getMessageId();
             mReferences = mInReplyTo;
+        } else if (K9.DEBUG) {
+            Log.d(K9.LOG_TAG, "could not get Message-ID.");
+        }
+
+        // Quote the message and setup the UI.
+        quotedMessagePresenter.processMessageToForward(messageViewInfo);
+        attachmentPresenter.processMessageToForward(messageViewInfo);
+    }
+
+    private void processMessageToForwardAsAttachment(MessageViewInfo messageViewInfo) throws IOException, MessagingException {
```

Comment:
This method is very similar to `processMessageToForward`, a `boolean forwardAsAttachment` parameter seems more suitable for the small difference in behavior